zerver/tests/test_thumbnail.py

```diff
@@ -587,14 +591,123 @@ def test_get_thumbnail(self) -> None:
                 path_id = re.sub(r"/user_uploads/", "", json_response["url"])
                 # Exit the block, triggering the thumbnailing worker
 
-            response = self.client_get(f"/user_uploads/thumbnail/{path_id}/{webp_still!s}")
-            self.assertEqual(response.status_code, 200)
-            self.assertEqual(response.headers["Content-Type"], "image/webp")
+            still_response = self.client_get(f"/user_uploads/thumbnail/{path_id}/{webp_still!s}")
+            self.assertEqual(still_response.status_code, 200)
+            self.assertEqual(still_response.headers["Content-Type"], "image/webp")
 
-            response = self.client_get(f"/user_uploads/thumbnail/{path_id}/{webp_anim!s}")
-            self.assertEqual(response.status_code, 404)
-            self.assertEqual(response.headers["Content-Type"], "image/png")
+            # We can request -anim, but we get the still webp because it's not animated
+            animated_response = self.client_get(f"/user_uploads/thumbnail/{path_id}/{webp_anim!s}")
+            self.assertEqual(animated_response.status_code, 200)
+            self.assertEqual(animated_response.headers["Content-Type"], "image/webp")
+            self.assertEqual(
+                animated_response.headers["Content-Length"],
+                still_response.headers["Content-Length"],
+            )
```

Comment:
I'm not entirely sure this part of the test reorganization is in the right commit.

Reply:
It used to be that we only got the still version, since that's all that we rendered.  We now _do_ get a 200 response to the animated version, because the "closest" logic kicks in and gives us the still version when we ask for the animated one -- hence the assertEqual on the file sizes.